TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/kafka/KafkaProtobufConsumer.java

```diff
@@ -0,0 +1,117 @@
+package org.opensearch.migrations.replay.kafka;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.opensearch.migrations.replay.ITrafficCaptureSource;
+import org.opensearch.migrations.trafficcapture.protos.TrafficStream;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+@Slf4j
+public class KafkaProtobufConsumer implements ITrafficCaptureSource {
+
+    public static final Duration CONSUMER_POLL_TIMEOUT = Duration.ofSeconds(1);
+    private final Consumer<String, byte[]> consumer;
+    private final String topic;
+    private final KafkaBehavioralPolicy behavioralPolicy;
+
+    public KafkaProtobufConsumer(Consumer<String, byte[]> consumer, String topic, KafkaBehavioralPolicy behavioralPolicy) {
+        assert topic != null;
+        this.consumer = consumer;
+        this.topic = topic;
+        this.behavioralPolicy = behavioralPolicy;
+    }
+
+    public static KafkaProtobufConsumer buildKafkaConsumer(String brokers, String topic, String groupId, boolean enableMSKAuth,
+        String propertyFilePath, KafkaBehavioralPolicy behavioralPolicy) {
+        if (brokers == null && topic == null && groupId == null) {
+            return null;
+        }
+        if (brokers == null || topic == null || groupId == null) {
+            throw new RuntimeException("To enable a Kafka traffic source, the following parameters are required " +
+                "[--kafka-traffic-brokers, --kafka-traffic-topic, --kafka-traffic-group-id]");
+        }
+        var kafkaProps = buildKafkaProperties(brokers, groupId, enableMSKAuth, propertyFilePath);
+        return new KafkaProtobufConsumer(new KafkaConsumer<>(kafkaProps), topic, behavioralPolicy);
+    }
+
+    public static Properties buildKafkaProperties(String brokers, String groupId, boolean enableMSKAuth, String propertyFilePath) {
```

Comment:
there are 3 string parameters.  Consider renaming this to include an indication as to what the order of the arguments should be.  buildKafkaParametersWithBrokerAndGroupId.  Even as I say that, I don't really like it - but I get nervous that brokers and groupId can be easily transposed (sometimes I'll make wrapper classes for added type safety).